File: easy/matrix-diagonal-sum/matrix-diagonal-sum.py
class Solution:
    def diagonalSum(self, mat: List[List[int]]) -> int:
        size = len(mat)
        sum = 0
        for i in range(size):
            sum += mat[i][i] + mat[i][size - i - 1]
        if size % 2 != 0:
            sum -= mat[size // 2][size // 2]
        return sum

File: easy/matrix-diagonal-sum/matrix-diagonal-sum.js
/**
 * @param {number[][]} mat
 * @return {number}
 */
var diagonalSum = function (mat) {
  const length = mat.length;
  let sum = 0;
  for (let i = 0; i < length; i++) {
    sum += mat[i][i] + mat[i][length - i - 1];
  }
  if (length % 2 != 0) {
    const mid = Math.floor(length / 2);
    sum -= mat[mid][mid];
  }
  return sum;
};
